Festergut Malleable Goo fixes and suggestions.

buza-me opened the pull request:
The changes were tested on PTR a couple weeks ago on the basic level in 25 heroic, to show the cooldown timer after Malleable Goo explodes. The timer values may be wrong. Code that can be improved is marked with TODO comments. I didn't try to go and fish for spell IDs and timers because you may already have the data.

broizter commented:
I very recently implemented a fix for Malleable Goo since the "trigger" used in the original DBM didn't work on this server. It now seem to work fine. Please take a look at it and tell if this PR is still relevant.

https://github.com/broizter/DBM-Whitemane/commit/712839a93f6de22ddb2b4ab16fdddff86cca86ab

broizter commented:
Ah sorry I completely missed that this was for Malleable Goo on Festergut and not Putricide. I will test this changes as soon as I get the chance, thanks!

buza-me commented:
Sure, whatever you think is best. If there is a timer people that want an alert can hook up a weakaura to a dbm trigger and get their alerts.

buza-me commented:
The CD timer and the alerts are separate things and only CD timer can be implemented.

broizter commented:
It seems like your code editor is setup to change the indentations automatically, making it difficult to review the changes since every line is showing up as changed

buza-me commented:
Sorry, I will fix it a bit later. There are no real changes except I moved goo warnings and timer into a local function, rebased the branch to main and fixed the conflicts.

broizter commented:
This mechanic only exists in heroic mode so it's only 10hc and 25hc we need spellids for right? I can find those out.

broizter commented:
Spell ID and variance on 25hc:
```
Malleable Goo-72550-npc:36678-3376618 = pull:26.55, 15.40, 17.22, 18.09, 18.30
```

broizter commented:
Please review the latest changes. Everything seem to work fine from my testing although the alert still sometimes trigger multiple seconds too early because of the variance of malleable goo cast.

- Hardcoded spell ids instead of grabbing the spellname
- Removed all malleable goo code from unit_spellcast_succeeded since this server does not generate an event there
- Formatted the code to be in line with how the rest of DBM code looks like

broizter commented:
Pushed a commit that disables the alert and timer by default. Can be enabled in Festergut settings for those who want.

broizter commented:
Merging. Please let me know if you feel like there is something that can be improved.

broizter commented:
@buza-me We killed Festergut on HC yesterday and I found that the timer and alert hurts more than it helps because of how unreliable it is. I have disabled it for now. Do some testing and make a new PR if you find a way to improve it.
